File: examples/withPostgreSQLDatabase/solutions/angularSpringBoot/backend/src/main/java/com/example/team/exampleproject3/service/BookService.java
package com.example.team.exampleproject3.service;

import javax.validation.Valid;

import com.example.team.exampleproject3.model.Book;
import com.example.team.exampleproject3.repository.BookRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class BookService {

	@Autowired
	private BookRepository repository;	

	public Iterable<Book> search() {
		return repository.findAll();
	}

	public Book getOne(Integer id) {
		return repository.findById(id).get();
	}

	public Book insert(Book book) {
		return repository.save(book);
	}

	public void update(@Valid Integer id, Book book) {
		repository.save(book);
	}

	public void delete(Integer id) {
		repository.deleteById(id);
	}

}

File: examples/withPostgreSQLDatabase/solutions/angularExpress/src/models/book/book.service.ts
import {Book} from './model/book.entity';

export class BookService {

    bookRepository = Book;

    findAll(): Promise<Book[]> {
        return this.bookRepository.findAll<Book>();
    }

    find(id: any): Promise<Book> {
        return this.bookRepository.findOne({where: {id}});
    }

    create(data: any): Promise<Book> {
        return data.save();
    }

    update(data: any): Promise<Book> {
        this.bookRepository.update<Book>(data, { where: {id: data.id}});
        return this.bookRepository.findOne({where: {id: data.id}});
    }

    delete(id: any): Promise<number> {
        return this.bookRepository.destroy({where: {id} });
    }
}
